src/main/java/io/cdap/plugin/http/common/http/OAuthUtil.java

```diff
@@ -16,86 +16,66 @@
 package io.cdap.plugin.http.common.http;
 
 import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.auth.oauth2.UserCredentials;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
-import com.google.gson.JsonElement;
 import io.cdap.plugin.http.common.BaseHttpConfig;
-import io.cdap.plugin.http.common.pagination.page.JSONUtil;
 import io.cdap.plugin.http.source.common.BaseHttpSourceConfig;
-import org.apache.http.client.methods.CloseableHttpResponse;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.utils.URIBuilder;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.util.EntityUtils;
+
 import java.io.ByteArrayInputStream;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URI;
-import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
 
 
 /**
  * A class which contains utilities to make OAuth2 specific calls.
  */
 public class OAuthUtil {
-  public static String getAccessTokenByRefreshToken(CloseableHttpClient httpclient, String tokenUrl, String clientId,
-                                                    String clientSecret, String refreshToken)
-    throws IOException {
-
-    URI uri;
-    try {
-      uri = new URIBuilder(tokenUrl)
-        .setParameter("client_id", clientId)
-        .setParameter("client_secret", clientSecret)
-        .setParameter("refresh_token", refreshToken)
-        .setParameter("grant_type", "refresh_token")
-        .build();
-    } catch (URISyntaxException e) {
-      throw new IllegalArgumentException("Failed to build token URI for OAuth2", e);
-    }
-
-    HttpPost httppost = new HttpPost(uri);
-    CloseableHttpResponse response = httpclient.execute(httppost);
-    String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
 
-    JsonElement jsonElement = JSONUtil.toJsonObject(responseString).get("access_token");
-    if (jsonElement == null) {
-         throw new IllegalArgumentException("Access token not found");
-    }
-    return jsonElement.getAsString();
+  public static OAuth2Credentials getUserCredentials(BaseHttpConfig config) {
+    UserCredentials userCredentials = new UserCredentials(config.getClientId(),
```

Comment:
I think this makes certain assumptions specific to google oauth apis. For example, it looks like it throws an exception if expires_in does not exist. However, not all providers return that.

Salesforce, for example, returns when the token was issued, but not when it expires (https://developer.salesforce.com/docs/atlas.en-us.196.0.api_rest.meta/api_rest/intro_understanding_refresh_token_oauth.htm). We can't use this, otherwise pipelines that were working before can start failing depending on their particular oauth response

Reply:
Good catch Albert.